### PR TITLE
Update documentation for GEN_KW

### LINKED ISSUE #7074: Update tab claims that parameter groups with all `CONST` are being localized, i.e., updated
**Describe the bug**

**To reproduce**
Steps to reproduce the behaviour:
1. Add new `GEN_KW` to `poly.ert` with all `CONST`.
   For example: `GEN_KW COEFFS_CONST coeff_priors_const` where the file `coeff_priors_const` contains:
```
a CONST 0
b CONST 2
c CONST 5
```
3. Run `Ensemble smoother` with `Adaptive Localization` which results in this screenshot:

image

**Expected behaviour**
Make it clear to users that we are not updating const-parameters.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Approach**
Some clarification of the documentation

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by dafeda** [comment]
This PR claims to resolve #7074 but I don't think that's the case.

**Comment by oyvindeide** [comment]
> This PR claims to resolve #7074 but I don't think that's the case.

So the problem here is that we have always updated these parameters, as we set const on a parameter level, and not the parameter group level, meaning that in a single parameter group you can have some parameters which are CONST and some which are not. So I propose that we fix this by documenting a way the users can exclude a parameter group entirely.

File: docs/reference/configuration/keywords.rst [hunk 1]
```diff
@@ -859,14 +859,51 @@ and/or history matching project.
...
 .. _gen_kw:
 .. topic:: GEN_KW
 
         The General Keyword, or :code:`GEN_KW` is meant used for specifying a limited number of parameters.
+        :code:`GEN_KW` supports either 2 or 4 positional arguments, as well as a few keyword arguments. The first
+        parameter is always the name of the parameter group. If given two positional arguments, those are:
+
+        ::
+
+                GEN_KW  <name of parameter>  <prior distribution file>
```
Comment (dafeda) [inline docs/reference/configuration/keywords.rst:875]:
```suggestion
                GEN_KW  <name of parameter group>  <prior distribution file>
```

File: docs/reference/configuration/keywords.rst [hunk 2]
```diff
@@ -859,14 +859,51 @@ and/or history matching project.
...
+
+
+        where:
+
+        | :code:`<name of parameter>` is an arbitrary unique identifier
```
Comment (dafeda) [inline docs/reference/configuration/keywords.rst:880]:
```suggestion
        | :code:`<name of parameter group>` is an arbitrary unique identifier
```

File: docs/reference/configuration/keywords.rst [hunk 3]
```diff
@@ -859,14 +859,51 @@ and/or history matching project.
...
+        | :code:`<prior distribution file>` is a file containing :ref:`parameter definitions <prior_distributions>`
+
+
+        In the case of 4 positional arguments, those are:
+
+        ::
+
+                GEN_KW  <name of parameter>  <template file> <output file on runpath> <prior distribution file>
```
Comment (dafeda) [inline docs/reference/configuration/keywords.rst:888]:
```suggestion
                GEN_KW  <name of parameter group> <template file> <output file on runpath> <prior distribution file>
```

File: docs/reference/configuration/keywords.rst [hunk 4]
```diff
@@ -859,14 +859,51 @@ and/or history matching project.
...
+
+        where:
+
+        | :code:`<name of parameter>` is an arbitrary unique identifier
```
Comment (dafeda) [inline docs/reference/configuration/keywords.rst:892]:
```suggestion
        | :code:`<name of parameter group>` is an arbitrary unique identifier
```

File: docs/reference/configuration/keywords.rst [hunk 5]
```diff
@@ -859,14 +859,51 @@ and/or history matching project.
...
+        | :code:`<output file on runpath>` name of the output file from ert containing templated values,
+        | :code:`<prior distribution file>` is a file containing :ref:`parameter definitions <prior_distributions>`
+
+        Keyword arguments:
+
+        ::
+
+                GEN_KW  ... UPDATE:TRUE/FALSE INIT_FILES:path/to/file_%d
+
+        where :code:`UPDATE` means if this parameter should be included in a history match run,
```
Comment (dafeda) [inline docs/reference/configuration/keywords.rst:903]: I suggest the following rephrasing that is perhaps a bit clearer:

Where the :code:`UPDATE` keyword argument specifies whether a parameter group should be included during the history matching process.
It must be set to either TRUE or FALSE.

**Review by dafeda — APPROVED** [review]

**Pull request merged**